extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -325,6 +353,11 @@ export class VisibilityManager {
       model.setReportReady(createReportPromiseFunc);
     }
 
+    const viewport = Services.viewportForDoc(this.ampdoc);
+    const scrollDepth = viewport.getScrollTop();
+    model.maybeSetInitialScrollDepth(scrollDepth);
```

Comment:
@jonkeller I believe what we want to do is to set initialscrollDepth only when the element is ready. 
So I think the logic looks like
```
if (readyPromise) {
  readyPromise.then(() => {
    //set initial scroll depth
  })
} esle {
  // immediately set initial scroll depth
}
```

Reply:
Thanks for the clarification - makes perfect sense. Done.